src-docs/src/views/resize_observer/resize_observer_hook.js

```diff
@@ -24,8 +24,8 @@ export const ResizeObserverHookExample = () => {
     setItems((items) => [...items, `Item ${items.length + 1}`]);
   };
 
-  const resizeRef = useRef();
-  const dimensions = useResizeObserver(resizeRef.current);
+  const [resizeRef, setResizeRef] = useState();
```

Comment:
❤️ Thanks @Juneezee!! The changes look terrific. Let's add a small inline comment here to note to devs that the usage of `useState` over `useRef` is intentional, and to help explain why!

```suggestion
  // Note: This must be a `useState` and not a `useRef` to correctly update on mount & unmount
  const [resizeRef, setResizeRef] = useState();
```